elliottlib/errata_async.py

```diff
@@ -16,13 +16,20 @@
 class AsyncErrataAPI:
     def __init__(self, url: str):
         self._errata_url = urlparse(url).geturl()
-        self._session = aiohttp.ClientSession(connector=aiohttp.TCPConnector(limit=32, force_close=True))
+        self._session = None
         self._gssapi_client_ctx = None
         self._headers = {
             "Content-Type": "application/json",
             "Accept": "application/json",
         }
 
+    async def _get_session(self, *args, **kwargs):
+        self._session = aiohttp.ClientSession(connector=aiohttp.TCPConnector(limit=32, force_close=True))
```

Comment:
aiohttp.ClientSession supports the context manager protocol for self closing. I think we should adopt this pattern instad of calling `session.close()` explicitly

Reply:
I didn't get you, do you mean we don't need to call session.close() here?https://github.com/openshift/elliott/pull/441/commits/521c7dc3e67f32231fb337ef1697d7792f5bd721#diff-52ba42ea595e4283138a9c20b65e159c6380a0b05856a3989bafb5777cba6a7dL52

Reply:
I referenced from here https://docs.aiohttp.org/en/stable/client_quickstart.html#make-a-request

>A session context manager usage is not mandatory but await session.close() method should be called in this case, e.g.:
```
session = aiohttp.ClientSession()
async with session.get('...'):
    # ...
await session.close()
```

Reply:
Sample from the docs:

```
import aiohttp
import asyncio

async def fetch(client):
    async with client.get('http://python.org') as resp:
        assert resp.status == 200
        return await resp.text()

async def main():
    async with aiohttp.ClientSession() as client:
        async with client.get('http://python.org') as resp:
            assert resp.status == 200
        print(resp.text())

loop = asyncio.get_event_loop()
loop.run_until_complete(main())
```

This way the session will close itself when done

Reply:
ok, updated